hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/HBaseThriftTestingUtility.java

```diff
@@ -0,0 +1,96 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.thrift;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.thrift.ThriftMetrics.ThriftServerType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HBaseThriftTestingUtility {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HBaseThriftTestingUtility.class);
+  private Thread thriftServerThread;
+  private volatile Exception thriftServerException;
+  private ThriftServer thriftServer;
+  private int testServerPort;
+
+  public int getServerPort() {
+    return testServerPort;
+  }
+
+  public void startThriftServer(Configuration conf, ThriftServerType t) throws Exception {
```

Comment:
nit: add a javadoc that this starts a thrift server based on 't'

Reply:
Done